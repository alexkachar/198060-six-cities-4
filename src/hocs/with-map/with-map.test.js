import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import withMap from './with-map.jsx';
import mockStore from '../../mocks/mock-store.js';
import mockOffers from '../../mocks/offers.js';

const MockComponent = ({mapRef}) => <div ref={mapRef}/>;
const WrappedComponent = withMap(MockComponent);

it(`Header renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <WrappedComponent offers={mockOffers} />
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});

MockComponent.propTypes = {
  mapRef: PropTypes.object.isRequired
};
