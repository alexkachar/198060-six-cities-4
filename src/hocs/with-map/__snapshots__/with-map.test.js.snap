// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMap renders correctly 1`] = `<div />`;
